eos/b-decays/b-to-d-l-nu.cc

```diff
@@ -80,26 +84,46 @@ namespace eos
         {
             double fp = form_factors->f_p(s);
             double f0 = form_factors->f_0(s);
+            double mbMSbar = model->m_b_msbar(mu);
             double lam = lambda(m_B * m_B, m_D * m_D, s);
-            double norm = power_of<2>(g_fermi())
-                / (384.0 * power_of<3>(M_PI * m_B));
-            // make sure we return NaN if s < m_l^2
-            double sqrtv = sqrt(1.0 - m_l * m_l / s);
-            double v = sqrtv * sqrtv, v2 = v * v;
-
-            // correct result
-            return norm * sqrt(lam) * v2 * ((3.0 - v) * fp * fp * lam + 3.0 * f0 * f0 * (1.0 - v) * power_of<2>(m_B * m_B - m_D * m_D));
+            double pBtoD = sqrt(lam) / (2.0 * m_B);
+            // make sure we return NaN if s < m_l^2, v = lepton velocity in the dilepton rest frame
+            double v = (1.0 - m_l * m_l / s);
+            double mvm1 = m_l * m_l / s;
+            double v2 = v * v;
+            double normBtoD = 8.0 * v2 * m_B * s * power_of<2>(g_fermi())
+                / (3.0 * 256.0 * power_of<3>(M_PI * m_B));
+            double hh0 = 2.0 * m_B * pBtoD * fp / (sqrt(s));
+            double hht = (m_B * m_B - m_D * m_D) * f0 / (sqrt(s));
+            //light quark masses neglected here
+            double hhs = (m_B * m_B - m_D * m_D) * f0 / (mbMSbar);
```

Comment:
All these variables are only assigned once. Please make them ```const double``` instead of ```double```, such that we cannot accidentally overwrite them.

Reply:
ok